server/server.go

```diff
@@ -116,7 +120,32 @@ func handleMountEvent(ctx context.Context, client *secretmanager.Client, creds c
 				Name: secret.ResourceName,
 			}
 			smMetricRecorder := csrmetrics.OutboundRPCStartRecorder("secretmanager_access_secret_version_requests")
-			resp, err := client.AccessSecretVersion(ctx, req, callAuth)
+			loc, err := locationFromSecretResource(secret.ResourceName)
+			if err != nil {
+				errs[i] = err
+				return
+			}
+
+			if len(loc) > locationLengthLimit {
+				errs[i] = fmt.Errorf("invalid location string, please check the location")
+				return
```

Comment:
Should we instead append to results[] and errs[]?

Reply:
I guess this will handle the race condition. Secret from ith index will wirte error or result on ith index only.